inc/components/class-head.php

```diff
@@ -41,18 +41,22 @@ public function set_from_query( $wp_query ) {
 		// Set default title.
 		$this->set_title( get_bloginfo( 'name' ) );
 
-		// If queried object is a valid article post type.
-		$queried_object = $wp_query->get_queried_object() ?? $wp_query->post;
-		if ( $queried_object instanceof \WP_Post ) {
-			$post_id = $queried_object->ID;
-
-			$this->apply_meta( $post_id );
-			$this->apply_social_meta( $post_id );
-			$this->set_title( $this->get_meta_title( $post_id ) );
-		} elseif ( $queried_object instanceof \WP_Term ) {
-			$this->set_title( $queried_object->name );
+		if( $wp_query->is_404() ) {
+			$this->set_title( '404 - Page not found' );
```

Comment:
You'll also want any text strings to be translated:
```suggestion
			$this->set_title( __( '404 - Page not found', 'wp-irving' ) );
```